File: misc/arithmetics/c/div_no_div/div_no_div.c
/**
 *
 * Division without division algorithm.
 * Notice that there is no control of zero or negative numbers or if b is greater than a.
 *
 */

#include <stdio.h>

int	main(void)
{
  int a = 15;
  int b = 3;
  int aux = a;
  int result = 0;

  while (aux)
  {
    result++;
    aux -= b;
  }

  printf("Result of %d * %d = %d\n", a, b, result);
  return (0);
}

File: misc/arithmetics/javascript/div_no_div/div_no_div.js
/**
 *
 * Division without division algorithm.
 * Notice that there is no control of zero or negative numbers or if b is greater than a.
 *
 */

const div_no_div = (a, b) => {

  	let aux = b;
	let i;

  	for(i = 0; aux <= a; i++)
	  aux += b;
		
	console.log(`Result of ${a} / ${b} = ${i}`);
}

div_no_div(30, 2);
